Automattic/woocommerce-services
# Update Webpack and related dependencies to the versions Calypso uses

In the spirit of using more and more parts of Calypso in WooCommerce Services, I updated most of the Webpack-related dependencies to the versions that Calypso uses, and made the Webpack config file more similar to Calypso's.

By updating to Webpack 4 and adopting some of the optimizations that Calypso uses (babel cache, multi-threading compilation, etc), the build is now much faster. In my 4-core Macbook Pro, after running `npm run dist` once (to prime the caches), the next times it takes around 7 seconds.

To test:
* Remember to run `npm ci` first, since a lot of the dependencies have changed
* Run `npm start`, do a smoke test on some of our UI
* Run `npm run dist`, do a smoke test on some of our UI (check that the bundle size didn't change significantly)
* Run `npm run i18n`, check that the `strings.php` file generated has the same strings (or similar, that command hasn't been run in a while)
